### PR TITLE
Fix Composer staging dockerimages

### PR DESCRIPTION
**Pull request by ravi-mosaicml**
#1197, which introduced composer image staging builds, installs Composer from pypi rather than from github. This PR fixes that by ensuring the modified buildArgs are passed to kaniko correctly.

### CODE REVIEW

File: .ci/Jenkinsfile
Comment (ravi-mosaicml) [inline .ci/Jenkinsfile:303]: TODO

File: .ci/Jenkinsfile
Comment (ravi-mosaicml) [inline .ci/Jenkinsfile:334]: TODO

File: .ci/Jenkinsfile
Comment (ravi-mosaicml) [reply .ci/Jenkinsfile]: removed

File: .ci/Jenkinsfile
Comment (ravi-mosaicml) [reply .ci/Jenkinsfile:332]:
```suggestion
                return
```

File: .ci/Jenkinsfile
Comment (ravi-mosaicml) [inline .ci/Jenkinsfile:332]: TODO
```suggestion
                return
```

File: docker/generate_build_matrix.py
Comment (ravi-mosaicml) [inline docker/generate_build_matrix.py:159]:
```suggestion
    # The `GIT_COMMIT` is a placeholder and Jenkins will substitute it with the actual git commit for the `composer_staging` images
```

File: .ci/Jenkinsfile
Comment (bandish-shah) [reply .ci/Jenkinsfile:330]: I'm going to approve to unblock you in the morning but please resolve this before merging.

**Review by bandish-shah — APPROVED** [review]
LGTM, please resolve the TODO's before merging.

File: .ci/Jenkinsfile
Comment (ravi-mosaicml) [reply .ci/Jenkinsfile]: Removed this TODO

**Pull request merged**
